### PR TITLE
Refine configurations files parsing

### LINKED ISSUE #194: Enhance pgmoneta_remove_whitespace function
# Enhancements in Comment Parsing

This issue highlights the improvements made in parsing comments to maintain the original format, including white-spaces, new lines, and tabs.

### Overview:

The old parsing method was primarily designed to remove all spaces in a string. This approach, often led to less readable comments. Moreover, applying this method to non-comment lines, or to entire file contents, or to commented code (debugging purposes), could result in syntax errors and damage the structure of the code.

### Examples:

1.  **Comments:**
    
    -   Original: `#int a = 10;`
    -   Parsed: `#inta=10;`
        -   _Issue:_ The removal of spaces merges the comment indicator with the code, making it less readable and causes syntax errors after uncomment it.

### Overview:

The new parsing method focuses on removing only leading and trailing spaces and lines, while preserving a single space between words. This approach is more refined and can be applied to entire files. It effectively parses comments without affecting readability or the actual code structure. In non-comment lines, it enhances readability by ensuring appropriate spacing.

### Key Features:

1.  **Preservation of Comment Structure:**
    
    -   Maintains the original format of comments, including spaces, tabs, and new lines.
2.  **Selective Space Removal:**
    
    -   Removes leading and trailing spaces and lines, but keeps a single space between words.
3.  **Applicability to Entire File:**
    
    -   The method is safe to apply to the entire file content, preserving the integrity of both comments and code.
4.  **Enhanced Readability:**
    
    -   In non-comment lines, the method improves readability by structuring spaces appropriately.

### PR DESCRIPTION
**Pull request by AhmedHisham99**
apply remove spaces from configurations lines to be parsed correctly

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Please, rebase.

You need to use the ` pgmoneta_remove_whitespace()` when loading the configuration files

**Comment by AhmedHisham99** [comment]
@jesperpedersen 
* update 
i have called the remove white space function between loading configuration line and processing.
can you please check this ?

File: src/libpgmoneta/configuration.c [hunk 1]
```diff
@@ -158,20 +159,21 @@ pgmoneta_read_configuration(void* shm, char* filename)
 
    while (fgets(line, sizeof(line), file))
    {
-      if (!is_empty_string(line))
+      parsed_line = pgmoneta_remove_whitespace(line);
```
Comment (jesperpedersen) [inline src/libpgmoneta/configuration.c:162]: You are missing `free(parsed_line)` for all paths, so you need to move it.

Use valgrind for checking

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgmoneta/configuration.c (hunk 1 above)
Comment (AhmedHisham99) [reply src/libpgmoneta/configuration.c:162]: yeah sure i will check now

**Comment by jesperpedersen** [comment]
Remember to squash your commits...

**Pull request closed without merging**